bin/download.tk1-exec-uboot

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2015, NVIDIA CORPORATION. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+if [ -n "${board_usb_dev}" ]; then
+    link=`readlink "${board_usb_dev}"`
+    instance=`echo "${link}" | sed -e "s@bus/usb@@"`
+    path="--usb-port-path ${instance}"
+fi
+
+"${download_tk1}/tegra-uboot-flasher" exec \
```

Comment:
The following command should be used here before calling `tegra-uboot-flasher exec`:

"${tubf_bin}/build" --boards "${tubf_board}" import-uboot "${U_BOOT_BUILD_DIR}"

Reply:
OK. What does that do and what does tubf stand for? I'd like to add some comments as I don't yet understand everything.

Reply:
"tubf" is "Tegra U-Boot Flasher".

"... exec" downloads code to the target from the copies in the build directory of the tubf project. "... import" copies files built elsewhere into the project in the location that "... exec" expects to find them. For a working example for TK1, see `bin/download.tegra-uboot-flasher`.

Reply:
OK thanks